src/account-multicall/LlamaAccountMulticallFactory.sol

```diff
@@ -0,0 +1,66 @@
+// SPDX-License-Identifier: MIT
+pragma solidity 0.8.23;
+
+import {LlamaAccountMulticallExtension} from "src/account-multicall/LlamaAccountMulticallExtension.sol";
+import {LlamaAccountMulticallGuard} from "src/account-multicall/LlamaAccountMulticallGuard.sol";
+import {LlamaAccountMulticallStorage} from "src/account-multicall/LlamaAccountMulticallStorage.sol";
+
+/// @title LlamaAccountMulticallFactory
+/// @author Llama (devsdosomething@llama.xyz)
+/// @notice This contract enables Llama instances to deploy an account multicall module.
+contract LlamaAccountMulticallFactory {
+  /// @dev Configuration of new Llama account multicall module.
+  struct LlamaAccountMulticallConfig {
+    address llamaExecutor; // The address of the Llama executor.
+    uint256 nonce; // The nonce of the new account multicall module.
+    LlamaAccountMulticallStorage.TargetSelectorAuthorization[] data; // The target-selectors to authorize.
+  }
+
+  /// @dev Emitted when a new Llama account multicall module is created.
+  event LlamaAccountMulticallModuleCreated(
+    address indexed deployer,
+    address indexed llamaExecutor,
+    uint256 nonce,
+    address accountMulticallGuard,
+    address accountMulticallExtension,
+    address accountMulticallStorage,
+    uint256 chainId
+  );
+
+  /// @notice Deploys a new Llama account multicall module.
+  /// @param accountMulticallConfig The configuration of the new Llama account multicall module.
+  /// @return accountMulticallGuard The deployed account multicall guard.
+  /// @return accountMulticallExtension The deployed account multicall extension.
+  /// @return accountMulticallStorage The deployed account multicall storage.
+  function deploy(LlamaAccountMulticallConfig memory accountMulticallConfig)
```

Comment:
We use `CREATE2` to have deterministic deployments of the 3 contracts.

Reply:
Decided against using minimal proxies here since the contract size is well within 24kb and it reduces complexity of the module and also doesn't include the additional `delegatecall` that comes with minimal proxies.